NeoML/src/Dnn/Layers/GELULayer.cpp

```diff
@@ -20,21 +20,41 @@ limitations under the License.
 
 namespace NeoML {
 
-static const float GELUMultiplier = 1.702f;
+// constants for the precise calculation and its backward
+static const float GELUOne = 1.0f;
+static const float GELUOneSecond = 0.5f;
+static const float GELUSqrt2Inv = 0.70710678f;
+static const float GELUSqrt2PiInv = 0.39894229f;
+
+// scale for the approximation
+static const float GELUApproximationMultiplier = 1.702f;
+
+static const int CGELULayerVersion = 1;
 
 CGELULayer::CGELULayer( IMathEngine& mathEngine ) :
 	CBaseLayer( mathEngine, "CGELULayer", false ),
-	multiplierVar( mathEngine )
+	oneVar( mathEngine ), oneSecondVar( mathEngine ), sqrt2InvVar( mathEngine ), sqrt2PiInvVar( mathEngine ), approxScaleVar( mathEngine )
 {
-	multiplierVar.SetValue( GELUMultiplier );
+	oneVar.SetValue( GELUOne );
+	oneSecondVar.SetValue( GELUOneSecond );
+	sqrt2InvVar.SetValue( GELUSqrt2Inv );
+	sqrt2PiInvVar.SetValue( GELUSqrt2PiInv );
+	approxScaleVar.SetValue( GELUApproximationMultiplier );
 }
 
-static const int CGELULayerVersion = 0;
-
 void CGELULayer::Serialize( CArchive& archive )
 {
-	archive.SerializeVersion( CGELULayerVersion );
+	const int version = archive.SerializeVersion( CGELULayerVersion );
```

Comment:
If you're loading old version of layer you must manually set the approximation method.
code sample::
```c++
geluLayer = new CGELULayer( mathEngine );
geluLayer->SetCalculationMode( CM_Precise );
geluLayr->Serialize( oldArchive ); // after this load calculation mode must be `CM_Sigmoid`
```